src/crypto/blake2b.c

```diff
@@ -286,7 +286,83 @@ int blake2b_init(blake2b_state *S, size_t outlen) {
 	return blake2b_init_param(S, &P);
 }
 
-int blake2b_init_key(blake2b_state *S, size_t outlen, const void *key, size_t keylen) {
+
+/**
+ * The difference from `blake2b_compress()` is that this function directly takes
+ * a variable-sized input, treating it as a block where the remaining bytes are
+ * zeros, and also wipes stack variables `m` and `v`. The intended use case is
+ * initializing hash states with keys, hence the stricter memory handling. I
+ * don't attempt to prevent swapping of the variables `m` and `v` since the OS
+ * would have to be dumber than a box of rocks to swap stack variables used just
+ * microseconds ago. `keylen` is assumed to be less than or equal to 128.
+ */
+static void blake2b_compress_key(blake2b_state *S, const uint8_t *key, size_t keylen) {
+	uint64_t m[16];
+	uint64_t v[16];
+	unsigned int i, r;
+
+	memset(m, 0, sizeof(m));
+#if defined(NATIVE_LITTLE_ENDIAN)
```

Comment:
To test the big-endian loading code, set this line to `#if 0`, compile, then run the hash tests